lib/elixir/lib/supervisor/spec.ex

```diff
@@ -75,6 +75,10 @@ defmodule Supervisor.Spec do
     * `:transient` - the child process is restarted only if it
       terminates abnormally, i.e., with an exit reason other than
       `:normal`, `:shutdown` or `{:shutdown, term}`
+  
+  Notice that supervisor that reached max restart intensity will exit with `:shutdown` reason. 
```

Comment:
I believe we only need the first sentence because people will be able to tie in to restart description they just read.

Or we should make the second sentence a bit clearer. Maybe: "In this case the supervisor will only be restarted if its child specification was defined with a `:restart` of `:permanent` (the default)." What do you think?